.circleci/integration-tests/master_dag.py

```diff
@@ -21,6 +21,17 @@
 MASTER_DAG_SCHEDULE = os.getenv("MASTER_DAG_SCHEDULE", None)
 IS_RUNTIME_RELEASE = bool(os.getenv("IS_RUNTIME_RELEASE", False))
 
+"""
+To run this master dag across multiple deployment parallelly, please set these env var unique per deployment
+
+GCP_BIGQUERY_DATASET_NAME,BATCH_JOB_COMPUTE_ENV, MODEL_NAME,AZURE_DATA_STORAGE_BLOB_NAME,AZURE_DATA_STORAGE_CONTAINER_NAME,
```

Comment:
I thought about whether we could add a single env var (e.g., `MASTER_DAG_RUN_SUFFIX`) that can work as a suffix to these variables so that we don't need to change all of it.

maybe something like the following?

```python
import os

MASTER_DAG_RUN_SUFFIX = os.getenv("MASTER_DAG_RUN_SUFFIX", None)

env_keys = [
    "GCP_BIGQUERY_DATASET_NAME",
    "BATCH_JOB_COMPUTE_ENV",
    "MODEL_NAME",
    "AZURE_DATA_STORAGE_BLOB_NAME",
    "AZURE_DATA_STORAGE_CONTAINER_NAME",
    "GCP_TEST_BUCKET",
    "GCP_DATAPROC_CLUSTER_NAME",
    "GKE_CLUSTER_NAME",
    "BATCH_JOB_NAME",
    "BATCH_JOB_QUEUE",
    "REDSHIFT_TABLE_NAME",
    "GKE_POD_NAME",
    "EMR_VIRTUAL_CLUSTER_NAME",
    "REDSHIFT_CLUSTER_IDENTIFIER",
    "EKS_CLUSTER_NAME",
    "EKS_NAMESPACE",
    "MANAGE_VIRTUAL_CLUSTERS",
    "JOB_EXECUTION_ROLE",
]

if MASTER_DAG_RUN_SUFFIX:
    for env_key in env_keys:
        value = os.getenv(env_key, None)
        if value:
            os.environ[env_key] += MASTER_DAG_RUN_SUFFIX

```

Reply:
After a second thought, this might not be sufficient. We'll need to pass these values to the triggered dag as well. Also it might be handy if the default value of `MASTER_DAG_RUN_SUFFIX` is some random value (maybe a timestamp?) so that all run is parallel by default

Reply:
Cloud resources usually have constraints on naming them, so we need to ensure that the timestamp format we wish to specify is accepted by all of them(usually the accept alphanumeric chars, but we observed some resources do not accept upper case characters 😄 ); otherwise resource creation would fail

Reply:
Yep, we can encode the timestamp into something acceptable by those resources

Reply:
Should we make it an issue if we're not going to implement this in this PR?